pkg/console/subresource/configmap/configmap.go

```diff
@@ -57,22 +61,49 @@ func getApiUrl(infrastructureConfig *configv1.Infrastructure) string {
 	return ""
 }
 
-func DefaultConfigMap(operatorConfig *operatorv1.Console, consoleConfig *configv1.Console, infrastructureConfig *configv1.Infrastructure, rt *routev1.Route) *corev1.ConfigMap {
-
+// DefaultConfigMap returns
+// - a new configmap,
+// - a bool indicating if config was merged (unsupportedConfigOverrides)
+// - an error
+func DefaultConfigMap(operatorConfig *operatorv1.Console, consoleConfig *configv1.Console, infrastructureConfig *configv1.Infrastructure, rt *routev1.Route) (*corev1.ConfigMap, bool, error) {
 	logoutRedirect := getLogoutRedirect(consoleConfig)
 	brand := getBrand(operatorConfig)
 	docURL := getDocURL(operatorConfig)
 	apiServerURL := getApiUrl(infrastructureConfig)
 
 	host := rt.Spec.Host
-	config := string(NewYamlConfig(host, logoutRedirect, brand, docURL, apiServerURL))
+	config := NewYamlConfig(host, logoutRedirect, brand, docURL, apiServerURL)
+
 	configMap := Stub()
-	configMap.Data = map[string]string{
-		consoleConfigYamlFile: config,
+	configMap.Data = map[string]string{}
+	unsupportedRaw := operatorConfig.Spec.UnsupportedConfigOverrides.Raw
+
+	// merge config overrides, if we have them
+	mergedConfig, err := resourcemerge.MergeProcessConfig(nil, config, unsupportedRaw)
+	if err != nil {
+		logrus.Errorf("failed to generate configmap: %v \n", err)
+		return nil, false, err
 	}
 
+	outConfigYaml, err := yaml2.JSONToYAML(mergedConfig)
+	if err != nil {
+		logrus.Errorf("failed to generate configmap: %v \n", err)
+		return nil, false, err
+	}
+
+	// if we actually merged config overrides, log this information
+	didMerge := len(operatorConfig.Spec.UnsupportedConfigOverrides.Raw) != 0
+	if didMerge {
+		logrus.Println(fmt.Sprintf("With UnsupportedConfigOverrides: %v", string(unsupportedRaw)))
+	}
+
+	configMap.Data[consoleConfigYamlFile] = string(outConfigYaml)
+
+	logrus.Println("generated console config yaml:")
+	logrus.Printf("%s", string(outConfigYaml))
```

Comment:
dont we need a newline ?
```suggestion
	logrus.Printf("%s \n", string(outConfigYaml))
```

Reply:
Yup